core/src/main/java/com/yahoo/imapnio/async/request/ExtendedSearchSequence.java

```diff
@@ -0,0 +1,75 @@
+package com.yahoo.imapnio.async.request;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.mail.search.SearchException;
+import javax.mail.search.SearchTerm;
+
+import com.sun.mail.iap.Argument;
+import com.sun.mail.imap.protocol.SearchSequence;
+import com.yahoo.imapnio.async.data.ExtendedModifiedSinceTerm;
+
+/**
+ * This class extends the search sequence for modification sequence with the optional fields entry
+ * name and type defined in https://tools.ietf.org/html/rfc7162#section-3.1.5.
+ */
+public class ExtendedSearchSequence extends SearchSequence {
+
+    /** Literal for MODSEQ. */
+    private static final String MODSEQ = "MODSEQ";
+
+    /**
+     * Generate the IMAP search sequence for the given search expression.
+     *
+     * @param term the search term
+     * @return the IMAP search sequence argument
+     * @throws SearchException when search too complex
+     * @throws IOException when failing to convert the given string into bytes in the specified charset
+     */
+    public Argument generateSequence(@Nonnull final SearchTerm term) throws SearchException, IOException {
+        if (term instanceof ExtendedModifiedSinceTerm) {
```

Comment:
The original Search Sequence only supports Modified Since Term and the original Modified Since Term only supports modification sequence. However, based on https://tools.ietf.org/html/rfc7162#section-3.1.5 , we also need [<entry-name> <entry-type-req>], I cannot override Modified Since Term since it is a final class. Therefore, I have to extend Search Term and extend Search Sequence to support this function.